src/Extension/Propagator/Jaeger/JaegerPropagator.php

```diff
@@ -0,0 +1,174 @@
+
+        $baggage = Baggage::fromContext($context);
+
+        if ($baggage->isEmpty()) {
+            return;
+        }
+
+        /** @var Entry $entry */
+        foreach ($baggage->getAll() as $key => $entry) {
+            $value = rawurlencode((string) $entry->getValue());
+            $setter->set(
+                $carrier,
+                sprintf('%s-%s', self::UBER_BAGGAGE_HEADER_PREFIX, $key),
+                $value
+            );
+        }
+    }
+
+    /** {@inheritdoc} */
+    public function extract($carrier, PropagationGetterInterface $getter = null, ContextInterface $context = null): ContextInterface
+    {
+        $getter ??= ArrayAccessGetterSetter::getInstance();
+        $context ??= Context::getCurrent();
+
+        $spanContext = self::extractImpl($carrier, $getter, $context);
+        if (!$spanContext->isValid()) {
+            return $context;
+        }
+
+        return $context->withContextValue(Span::wrap($spanContext));
+    }
+
+    private function getFlag(SpanContextInterface $spanContext, ContextInterface $context): string
+    {
+        if ($spanContext->isSampled()) {
+            if ($context->get(JaegerDebugFlagContextKey::instance())) {
+                return self::IS_DEBUG;
+            }
+
+            return self::IS_SAMPLED;
+        }
+
+        return self::IS_NOT_SAMPLED;
+    }
+
+    private static function extractImpl($carrier, PropagationGetterInterface $getter, ContextInterface &$context): SpanContextInterface
+    {
+        $headerValue = $getter->get($carrier, self::UBER_TRACE_ID_HEADER);
+
+        if ($headerValue === null) {
+            return SpanContext::getInvalid();
+        }
+
+        $pieces = explode(':', $headerValue);
+
+        if (count($pieces) != 4) {
+            return SpanContext::getInvalid();
+        }
+
+        [$traceId, $spanId, $parentSpanId, $traceFlags] = $pieces;
+
+        $traceId = str_pad($traceId, SpanContextValidator::TRACE_LENGTH, '0', STR_PAD_LEFT);
+        $spanId = str_pad($spanId, SpanContextValidator::SPAN_LENGTH, '0', STR_PAD_LEFT);
+
+        if (!SpanContextValidator::isValidTraceId($traceId) || !SpanContextValidator::isValidSpanId($spanId)) {
+            return SpanContext::getInvalid();
+        }
+
+        if ($traceFlags === self::IS_DEBUG) {
+            $context = $context->with(JaegerDebugFlagContextKey::instance(), true);
+        }
+
+        $context = self::extractBaggage($carrier, $context);
+
+        $isSampled = ((int) $traceFlags) & 1;
+
+        return SpanContext::createFromRemoteParent(
+            $traceId,
+            $spanId,
+            $isSampled ? TraceFlags::SAMPLED : TraceFlags::DEFAULT
+        );
+    }
+
+    private static function extractBaggage($carrier, ContextInterface $context): ContextInterface
+    {
+        $baggageBuilder = Baggage::getBuilder();
+
+        foreach ($carrier as $key => $value) {
+            if (strpos($key, self::UBER_BAGGAGE_HEADER_PREFIX) === 0) {
```

Comment:
Could use `str_starts_with()`.
Also doesn't seem to handle the `-` after `uberctx`.

Reply:
This function is only available for PHP 8, so the CI for version 7.4 is failing.